dataloader/loader.py

```diff
@@ -113,12 +124,13 @@ def load_segment(
     return json_segment["segnr"]
 
 
-def load_files_collection(file: MenuItem, segmentnrs: list, connection: Connection):
+def load_files_collection(file: MenuItem, segmentnrs: list, parallelcount: list, connection: Connection):
     collection = connection[COLLECTION_FILES]
     doc = collection.createDocument()
     doc._key = file["filename"]
     doc.set(file)
     doc["segmentnrs"] = segmentnrs
+    doc["parallelcount"] = parallelcount
```

Comment:
1. Yes, but not for the graphs. I made this first and then remembered that the graphs need to have the filters employed. But I left the code there because I need it for the visual charts later.
2. Putting `filename` in there was your idea in order to be able to filter on it instead of filtering something that needed to be calculated from the `_key`. 
You could in theory also put a long list of parallel-id in there and then find that file data, go to each of those parallel-id and then see if that parallel fits with the filters and if yes, add it. But I think that would add more steps in AQL and might slow it down. The list of parallel-id is already per segment so in theory you can also now first filter for the right filename in the collection `files` then extracty the segment numbers, go to each of those in collection `segments`, and from there go to each of the parallels. I just thought it would be far easier to just go to the one collection and retrieve everything I need from there.